### PR TITLE
Bug 1366936: fix ICT matching in the trigger controller

### PR DESCRIPTION
**Pull request by kargakis**
@mfojtik @smarterclayton 

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1366936

### CODE REVIEW

**Comment by kargakis** [comment]
[test]

**Comment by mfojtik** [comment]
LGTM

**Comment by kargakis** [comment]
quota and https://github.com/openshift/origin/issues/9856 [test]

**Comment by kargakis** [comment]
#9856 [test]

**Comment by kargakis** [comment]
Adding an extended test maybe shortly, maybe tomorrow

File: test/extended/deployments/deployments.go [hunk 1]
```diff
@@ -242,6 +243,31 @@ var _ = g.Describe("deploymentconfigs", func() {
...
 
+	g.Describe("with multiple image change triggers", func() {
+		g.AfterEach(func() {
+			failureTrap(oc, "logging-kibana", g.CurrentGinkgoTestDescription().Failed)
+		})
+
+		g.It("should run a successful deployment [Conformance]", func() {
+			_, err := oc.Run("create").AsAdmin().Args("-f", loggingDeployerFixture).Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			_, err = oc.Run("new-app").AsAdmin().Args("logging-deployer-account-template").Output()
```
Comment (kargakis) [inline test/extended/deployments/deployments.go]: @deads2k @mfojtik I cannot seem to create cluster scoped resources with `AsAdmin`. Do we run other extended tests where we test cluster-scoped stuff?

Comment (deads2k) [inline test/extended/deployments/deployments.go]: > @deads2k @mfojtik I cannot seem to create cluster scoped resources with AsAdmin. Do we run other extended tests where we test cluster-scoped stuff?

I don't know.

**Comment by kargakis** [comment]
[test]

**Comment by kargakis** [comment]
extended test added, [merge]

**Pull request merged**